crates/typst-library/src/foundations/selector.rs

```diff
@@ -221,6 +226,15 @@ impl Selector {
             inclusive,
         }
     }
+
+    /// Returns a modified selector that only matches `self` if it is contained in an `ancestor`.
```

Comment:
Suggestion: Directly note that the matching is done flatly:

```suggestion
    /// Returns a modified selector that only matches `self` if it is directly contained in an `ancestor`.
```

Reply:
it doesn't, but i apparently *do* need to dedupe (managed to get duplicate results using nested figures)

Reply:
Fair, should probably reword then: Explicitly denote the depth behavior